magicctapipe/image/tests/test_calib.py

```diff
@@ -0,0 +1,352 @@
+from magicctapipe.image.calib import calibrate
+import pytest
+
+
+
+from ctapipe.calib import CameraCalibrator
+
+from ctapipe.io import EventSource
+
+from magicctapipe.image import MAGICClean
+from traitlets.config import Config
+
+
+@pytest.fixture(scope="session")
+def tel_id_LST():
+    return 1
+
+
+@pytest.fixture(scope="session")
+def tel_id_MAGIC():
+    return 2
+
+
+def test_calibrate_LST(dl0_gamma, config_calib, tel_id_LST):
+    
+    assigned_tel_ids = [1,2,3]
+    for input_file in dl0_gamma:
+        event_source = EventSource(
+            input_file,
+            allowed_tels=assigned_tel_ids,
+            focal_length_choice='effective'
+        )
+
+
+        obs_id = event_source.obs_ids[0]
+
+        subarray = event_source.subarray    
+
+        tel_descriptions = subarray.tel   
+        camera_geoms={}
+
+        for tel_id, telescope in tel_descriptions.items():
+            camera_geoms[tel_id]= telescope.camera.geometry        
+       
+        config_lst = config_calib["LST"]
+
+        extractor_type_lst = config_lst["image_extractor"].pop("type")
```

Comment:
if you do not "pop" the  extractor type here, but just access it as config_lst["image_extractor"]["type"] you will not need to reintroduce it in line 74

Reply:
While implementing these functions (then merged in a single function) we had to use "pop" because the dictionary Is then passed to lstchain functions that will complain if they meet the entry "type" in their dictionary. But I can check this again and let you know

Reply:
I checked this again, if we don't use 'pop' the lstchain functions will complain (e.g., `TypeError: add_noise_in_pixels() got an unexpected keyword argument 'use'`). I don't know if these functions have been updated in the last lstchain version, but maybe we can ask the developers to change a bit the allowed arguments (if possible)

Reply:
I am confused at why the error mentions a 'use' keyword. It doesn't relate to the "image_extractor".  But indeed the next line requires 'type' to not be here to have on lstchain like config. Alternatively we could modify our yaml file in one of the following ways:
- Like in lstchain, have on image extractor field for the name, and a config field for the associated extractor : 
```
 "image_extractor": "LocalPeakWindowSum",
  "LocalPeakWindowSum":{
    "window_shift": 4,
    "window_width": 8,
    "apply_integration_correction": false
  },
```
- have a type and config field:
```
LST:
    image_extractor:
        type: "LocalPeakWindowSum"
        config: window_shift: 4
                     window_width: 8
```

Reply:
I just reported an error due to not using the 'pop' in one of the lines mentioned by Julian (this was in the NSB correction, but analogous in all the cases) just not to divide the conversation in a lot of scattered pieces (We can discuss the issue in the first conversation and the same solution will be adopted to solve all of the related conversations). BTW, yes, we could change the config file structure, if everybody agrees

Reply:
ok, lstchain-like solution is fine with me. It would be problematic only if we want to use the same extractor twice with different parameters for two different purposes (like regular charge extraction and something special for muons), but since this is not an issue in LST, it should not affect MCP either. 
But if we update it, we should be careful not to leave in the code any place where this config is looked for in the wrong place (and here again the muon script might behave differently). Since all this was just to exclude a few lines of code and it turned out to be much more tricky than thought in the beginning I think the simplest solution would be to just leave it as it is for now and come back to it in another PR (maybe after lstchain/ctapipe version update)

Reply:
I agree with Julian: We have to check again each script and function using this configuration parameters, so maybe we can open a small PR after we finish these big updates, or at least after a pipeline cleanup (the smaller the repository, the easier will be not to forget some changes). In the meanwhile, we should check the lstchain functions' upgrades because with ctapipe 0.19 we have to switch to lstchain 0.10 and maybe we have to further update the configuration file

Reply:
ok, so closing this now...